tests/integration/Mailer/WordPress/WordpressMailerTest.php

```diff
@@ -268,6 +275,131 @@ public function testItAddSubscriberMetaInfo() {
     $wpMailer->send();
   }
 
+  /*
+   * Test for issue https://mailpoet.atlassian.net/browse/MAILPOET-3707
+   */
+  public function testItPreservesWPReplyTo() {
+    list($mailer, $fallbackMailer) = $this->getMailerInstancesForReplyToTests();
+
+    $wpMailer = new WordPressMailer($mailer, $fallbackMailer, new MetaInfo, $this->subscribersRepository);
+    $wpMailer->addAddress('email@example.com', 'Full Name');
+    $wpMailer->addReplyTo('reply-to@example.com', 'Reply To');
+    $wpMailer->From = 'email-from@example.com';
+    $wpMailer->Body = 'Body';
+    $wpMailer->send();
+
+    $mailer = $this->getPrivateProperty($wpMailer, 'mailer');
```

Comment:
Similar to what I mentioned above, I don't like this part of the tests that I wrote as well, but I settled with it, for now, to not spend too much time on this task before getting some feedback.

I had to create a hack using reflection to get the value of an private property to check if the reply-to address is correct, which is an anti-pattern for tests. But again, I don't see another way to write the tests without refactoring the Mailer class.